twilix/patterns/__init__.py

```diff
@@ -0,0 +1,23 @@
+from twisted.internet import task
+
+from twilix.jid import internJID
+
+class BasePattern(object):
+    def __init__(self, myjid, keepalive_period=None):
+        """ :param myjid: The jid of the component."""
+
+        self.myjid = internJID(myjid)
+        if keepalive_period is not None:
+            self.keepalive_period = keepalive_period
```

Comment:
I believe it should be easier to set it unconditionally. Later you won't need hasattr.